core/src/sql/array.rs

```diff
@@ -498,3 +498,27 @@ impl Uniq<Array> for Array {
 		self
 	}
 }
+
+// ------------------------------
+
+pub(crate) trait Windows<T> {
+	fn windows(self, window_size: usize) -> Result<T, Error>;
+}
```

Comment:
Do we need a trait for this, it's only implemented in one place right?

Reply:
We do this so that we don't pollute the `Array` type itself with lots of functions. If you want the function, you need to import the trait!